File: 187-repeated-dna-sequences/repeated-dna-sequences.py
class Solution(object):
    def findRepeatedDnaSequences(self, s):
        """
        :type s: str
        :rtype: List[str]
        """
        ans = []
        mpp = {}
        if len(s) < 10:
            return ans
        for i in range(len(s) - 9):
            temp = s[i:i + 10] 
            mpp[temp] = mpp.get(temp, 0) + 1
        for key, value in mpp.items():
            if value > 1:
                ans.append(key)
        return ans
        

File: 187-repeated-dna-sequences/repeated-dna-sequences.cpp
class Solution {
public:
    vector<string> findRepeatedDnaSequences(string s) {
        vector<string>ans;
        int i,j;
        map<string,int>mpp;
        if(s.size()<10)return ans;
        for(i=0;i<=s.size()-10;i++){
            string temp = s.substr(i,10);
            mpp[temp]++;
        }
        for(auto x : mpp){
            if(x.second>1){
                ans.push_back(x.first);
            }
        }
        return ans;
    }
};